packages/ra-ui-materialui/src/button/SaveButton.spec.tsx

```diff
@@ -41,11 +45,53 @@ describe('<SaveButton />', () => {
                 />
             </TestContext>
         );
-
         fireEvent.click(getByLabelText('ra.action.save'));
         expect(onSubmit).toHaveBeenCalled();
     });
 
+    it('should trigger submit action when Enter key if no saving is in progress', () => {
+        const onSubmit = jest.fn();
+
+        const defaultCreateProps = {
+            basePath: '/foo',
+            id: '123',
+            resource: 'foo',
+            location: {},
+            match: {},
+        };
+
+        const SaveToolbar = props => (
+            <Toolbar {...props}>
+                <SaveButton
+                    label="enterButton"
+                    handleSubmitWithRedirect={onSubmit}
+                    submitOnEnter={true}
+                    saving={false}
+                />
+            </Toolbar>
+        );
+
+        const { getByLabelText } = render(
+            <TestContext>
+                <Create {...defaultCreateProps}>
```

Comment:
You don't need a Create to test the SaveButton. It's also broken when using directly the SimpleForm.

Reply:
I tried various things, the test never pass.
I don't know why
Hope that'll do it in e2e